`protostar declare` (#459)
software-mansion/protostar: website/docs/tutorials/guides/deploying.md

# Deploying

## Prerequisites
- [Factory Contract pattern](https://research.csiro.au/blockchainpatterns/general-patterns/contract-structural-patterns/factory-contract/)
- [Deploying a contract by another contract](https://www.cairo-lang.org/docs/hello_starknet/deploying_from_contracts.html?highlight=class%20hash)

> ## Declaring a contract
> StarkNet separates contracts into classes and instances. To create a new contract, you need to:
> 1. declare a class
> 2. call the deploy syscall with the `class_hash` from the declare transaction as an argument
>    ```cairo
>    from starkware.starknet.common.syscalls import deploy
>    ```
>    
> To declare a contract from the Protostar CLI you need to build your project and use the `protostar declare` command.
> 

Reviewer: ```suggestion
## Declaring a contract
StarkNet provides a distinction between contract class and instance. To create a new contract, you need to:
1. Declare a contract
2. Use the deploy syscall with the `class_hash` from the declare transaction as an argument
   ```cairo
   from starkware.starknet.common.syscalls import deploy
   ```
   
To declare a contract from the Protostar CLI you need to build your project and use the `protostar declare` command.

```

```sh
$ protostar declare ./build/main.json --network alpha-goerli
```

> ```sh title="The result of running 'protostar declare'."

Author: Nope. Removed.
